src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -122,6 +124,7 @@ protected ShardIterator shards(ClusterState clusterState, InternalRequest reques
 
         long[] preVersions = new long[request.items().length];
         VersionType[] preVersionTypes = new VersionType[request.items().length];
+        Translog.Location location = null;
         for (int requestIndex = 0; requestIndex < request.items().length; requestIndex++) {
```

Comment:
No that does not make sense, please, add hard checks or asserts. This explanation nor a comment explaining that logic is enough. We should check that the translog file is not changing during this loop and that no positions go backwards, or we should not do it like this at all.